### PR TITLE
Fix nested module "unknown variable" during dest

### PR DESCRIPTION
**Pull request by jbardin**
During a destroy with nested modules, access a variable between them
causes an "unknown variable accessed" during destroy.

### CODE REVIEW

File: terraform/test-fixtures/apply-destroy-nested-module-with-attrs/middle/middle.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+variable param {}
```
Comment (phinze) [inline terraform/test-fixtures/apply-destroy-nested-module-with-attrs/middle/middle.tf]: technically valid to leave the var name bare, but conventionally we always quote it -> `variable "param" {}`

File: terraform/test-fixtures/apply-destroy-nested-module-with-attrs/main.tf [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (phinze) [inline terraform/test-fixtures/apply-destroy-nested-module-with-attrs/main.tf]: This file can be dropped, nothing requires a `main.tf` to exist.

File: terraform/context_apply_test.go [hunk 3]
```diff
@@ -4704,3 +4704,84 @@ aws_instance.foo:
...
+		ctx := testContext2(t, &ContextOpts{
+			Destroy: true,
+			Module:  m,
+			State:   state,
+			Providers: map[string]ResourceProviderFactory{
+				"null": testProviderFuncFixed(p),
+			},
+		})
+
+		// First plan and apply a create operation
```
Comment (phinze) [inline terraform/context_apply_test.go]: This comment no longer accurate - this is the destroy now.

Comment (jbardin) [inline terraform/context_apply_test.go]: I think this was for another comment (I've since removed)? This block is doing a Plan+Apply.

Comment (phinze) [inline terraform/context_apply_test.go]: Oh weird - yeah last time I made this comment it was on the below block. Weird diff hunk issue maybe.

File: terraform/graph_config_node_variable.go [hunk 4]
```diff
@@ -130,17 +130,22 @@ func (n *GraphNodeConfigVariable) hasDestroyEdgeInPath(opts *NoopOpts, vertex da
 	if vertex == nil {
 		vertex = opts.Vertex
 	}
+
+	var root graphNodeRoot
```
Comment (phinze) [inline terraform/graph_config_node_variable.go]: I see how this works, but it feels like a bit of an odd way to test if a node is the root node. Maybe worth wrapping this up in a `IsGraphNodeRoot()` predicate stored near the `graphNodeRoot` impl?

Or here's another idea - IMHO this impl is a bit less confusing than the local var / `==` test:

```go
_, isRoot := v.(graphNodeRoot)
if isRoot || cv.DestroyEdgeInclude(v) { // ...
```

What do you think?

Comment (jbardin) [inline terraform/graph_config_node_variable.go]: Yeah, checking that c is a root node is a little cleaner. Unless we had a single global rootNode value to compare to, I agree the struct comparison felt weird. I'll update

**Comment by phinze** [comment]
LGTM

**Comment by jbardin** [comment]
This fixes #7378

**Pull request merged**